cpp/tests/geometry/PointCloud.cpp

```diff
@@ -876,13 +876,15 @@ TEST(PointCloud, FarthestPointDownSample) {
                               {1.0, 0, 1.0},
                               {0, 1.0, 1.0},
                               {1.0, 1.0, 1.5}});
+    std::vector<Eigen::Vector3d> expected = {
+            {0, 2.0, 0}, {1.0, 1.0, 0}, {1.0, 0, 1.0}, {0, 1.0, 1.0}};
     std::shared_ptr<geometry::PointCloud> pcd_down =
             pcd.FarthestPointDownSample(4);
-    ExpectEQ(pcd_down->points_, std::vector<Eigen::Vector3d>({{0, 2.0, 0},
-                                                              {1.0, 1.0, 0},
-                                                              {1.0, 0, 1.0},
-                                                              {0, 1.0, 1.0}}));
-}  // namespace tests
+    std::shared_ptr<geometry::PointCloud> pcd_down_2 =
+            pcd.FarthestPointDownSample(4, 0);
```

Comment:
You can compute the result for non-zero start_index with the function and then use this in the unit test. I think this is ok here because there were no major changes to the algorithm.